src/jit/codegencommon.cpp

```diff
@@ -2754,14 +2754,12 @@ void CodeGen::genExitCode(BasicBlock* block)
 
 void CodeGen::genJumpToThrowHlpBlk(emitJumpKind jumpKind, SpecialCodeKind codeKind, GenTreePtr failBlk)
 {
-    bool useThrowHlpBlk = !compiler->opts.compDbgCode;
-
 #if defined(UNIX_X86_ABI) && FEATURE_EH_FUNCLETS
     // Inline exception-throwing code in funclet to make it possible to unwind funclet frames.
     useThrowHlpBlk = useThrowHlpBlk && (compiler->funCurrentFunc()->funKind == FUNC_ROOT);
 #endif // UNIX_X86_ABI && FEATURE_EH_FUNCLETS
 
-    if (useThrowHlpBlk)
+    if (compiler->fgUseThrowHelperBlocks())
```

Comment:
This won't compile for Unix x86 with Funclets: the above code adjusts `useThrowHlpBlk`.

Reply:
While you're here, maybe you can update the function header with the new standard header, and describe the arguments.

Reply:
There are various comments here about "for debuggable code" and "for non-debuggable code", which seem to be wrong at this point, since fgUseThrowHelperBlocks() might be more complex than that.